weather/www/js/services.js

```diff
@@ -0,0 +1,214 @@
+/* Notes on forecast.io's API:
+ *  - daily.data[0] is today
+ * By: Cody Davey & Travis Pullen
+ */
+
+/* weatherStore holds data for weatherData service.
+ */
+weatherApp.factory('weatherStore', function() {
+    var weatherStore = {
+        current : {}
+    };
+
+    return weatherStore;
+});
+
+/* weatherData service gets weather data (current, forecasts, and
+ * historical), processes it as necessary, and provides it to controllers.
+ */
+weatherApp.service('weatherData', ['$q', '$resource', '$http',
+                                   'FORECASTIO_KEY', 'weatherStore',
+    function($q, $resource, $http, FORECASTIO_KEY, weatherStore) {
+        this.getCurrentWeather = function(lat, lng) {
+            var url = 'https://api.forecast.io/forecast/' +
+                FORECASTIO_KEY + "/" + lat + ',' + lng;
+
+            // JSONP is only needed for "ionic serve".
+            // Simpler $http.get(url) works on devices.
+            return $http.jsonp(url + '?callback=JSON_CALLBACK').then(
+                function success(resp) {
+                    weatherStore.current = resp.data;
+                    console.log('GOT CURRENT');
+                    console.dir(weatherStore.current);
+                },
+                function failure(error) {
+                    alert('Unable to get current conditions');
+                    console.error(error);
+                });
+        };
+
+        this.getLocation = function() {
+            return $q(function(resolve, reject) {
+                navigator.geolocation.getCurrentPosition(function(position) {
+                    resolve({
+                        latitude: position.coords.latitude,
+                        longitude: position.coords.longitude
+                    });
+                }, function(err) {
+                    reject(err);
+                });
+            });
+        };
+
+        // Round temp to tenths of a degree.
+        this.roundTemp = function(temp) {
+            if (Math.abs(temp) >= 10) {
+                return temp.toPrecision(3);
+            } else {
+                return temp.toPrecision(2);
+            }
+        };
+
+        // Return current temperature
+        this.tempNow = function() {
+            return this.roundTemp(weatherStore.current.currently.temperature);
+        };
+
```

Comment:
Yep! This was the initial plan, but time constraints led to me taking the short and easy, though also messy, way out. I'll do this instead.